src/input.cpp

```diff
@@ -111,6 +111,11 @@ void InputHandler::handleEvent(sf::Event& event)
             fireKeyEvent(last_key_press, event.text.unicode);
             last_key_press.code = sf::Keyboard::Unknown;
         }
+        else
+        {
+            // No relevant code for a KeyEvent, so pass the TextEvent instead.
```

Comment:
I think it would be better to split the fireKeyEvent into the key_code and text event versions, instead of having a special case for just text events that have no valid keycode. Yes, this will require quite a few changes in EE :)